custom_model_runner/datarobot_drum/drum/drum.py

```diff
@@ -64,56 +66,150 @@ def __init__(self, runtime):
+            return
 
-        if hasattr(self.options, "target_type") and self.options.target_type is not None:
-            target_type_options = TargetType(self.options.target_type)
+        target_type_options = getattr(self.options, "target_type", None)
+        target_type_options = (
+            None if target_type_options is None else TargetType(target_type_options)
+        )
+        target_type_model_config = None
 
         if self.options.model_config is not None:
             target_type_model_config = TargetType(self.options.model_config["targetType"])
 
-        if self.run_mode not in [RunMode.NEW]:
-            if target_type_options is None and target_type_model_config is None:
+        if target_type_options is None and target_type_model_config is None:
+            raise DrumCommonException(
+                "Target type is missing. It must be provided in --target-type argument, {} env var or model config file.".format(
+                    ArgumentOptionsEnvVars.TARGET_TYPE
+                )
+            )
+        elif (
+            all([target_type_options, target_type_model_config])
+            and target_type_options != target_type_model_config
+        ):
+            raise DrumCommonException(
+                "Target type provided in --target-type argument doesn't match target type from model config file. "
+                "Use either one of them or make them match."
+            )
+        else:
+            self.target_type = (
+                target_type_options if target_type_options is not None else target_type_model_config
+            )
+
+        if self.target_type != TargetType.UNSTRUCTURED:
+            if getattr(self.options, "query", None):
+                raise DrumCommonException(
+                    "--query argument can be used only with --target-type unstructured"
+                )
+            if getattr(self.options, "content_type", None):
+                raise DrumCommonException(
+                    "--content-type argument can be used only with --target-type unstructured"
+                )
+        else:
+            if self.options.content_type is None:
+                self.options.content_type = "text/plain; charset=utf8"
+
+    def _resolve_class_labels(self):
+        if (
+            self.run_mode in [RunMode.NEW]
+            or self.run_mode == RunMode.PUSH
+            and self.options.model_config[ModelMetadataKeys.TYPE] == "training"
+        ):
+            self.options.positive_class_label = None
+            self.options.negative_class_label = None
+            self.options.class_labels = None
+            self.options.class_labels_file = None
+            return
+
+        if self.target_type == TargetType.BINARY:
+            pos_options = getattr(self.options, "positive_class_label", None)
+            neg_options = getattr(self.options, "negative_class_label", None)
+
+            try:
+                pos_model_config = self.options.model_config.get(
```

Comment:
you don't need .get here.

Reply:
This is the way to avoid using a check `if key in dict`:
If key is absent get returns None. Trying to call get on None causes AttributeError.

Reply:
Consider the case: model type is training, but has Inference section.... what should we do?
Because inference section is the same as providing labels in CLI.